src/libcollections/str.rs

```diff
@@ -764,14 +937,15 @@ impl str {
         core_str::StrExt::find(self, pat)
     }
 
-    /// Returns the byte index of the last character of `self` that
-    /// matches the pattern, if it
-    /// exists.
+    /// Returns the byte index of the last character of this string slice that
+    /// matches the pattern.
+    ///
+    /// Returns `None` if the pattern doesn't match.
     ///
-    /// Returns `None` if it doesn't exist.
+    /// The pattern can be a `&str`, [`char`], or a closure that determines if
+    /// a character matches.
```

Comment:
Do you expect such documentation to be easily upgradeable when more things start implementing `Pattern`? It doesn’t even list all the current `Pattern` implementors.